torch/distributed/checkpoint/logger.py

```diff
@@ -0,0 +1,92 @@
+import copy
+import functools
+import time
+from typing import Any, Callable, Dict, List, Tuple, TypeVar
+
+from typing_extensions import ParamSpec
+
+import torch.distributed.c10d_logger as c10d_logger
+from torch.distributed.checkpoint.logging_handlers import DCP_LOGGER_NAME
+
+__all__: List[str] = []
+
+global _dcp_logger
+_dcp_logger = c10d_logger._get_or_create_logger(DCP_LOGGER_NAME)
+
+_T = TypeVar("_T")
+_P = ParamSpec("_P")
+
+
+def _parse_dcp_method_args(*args, **kwargs) -> Tuple[Tuple[Any], Dict[str, Any]]:
```

Comment:
Since this is eventually used in a functional wrapper, you can use `inspect` to inspect the function arguments and keyword arguments. This is just a suggestion and you don't have to change it in this PR. And this may be useful if we want to use the same logger wrapper for distributed state_dict.

Reply:
Good idea, thanks!